curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-mixed-grouping-of-characters.english.md

```diff
@@ -0,0 +1,66 @@
+---
+id: 5c3dda8b4d8df89bea71600f
+title: Check For Mixed Grouping of Characters
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+Sometimes we want to check for groups of characters using a Regular Expression, to achieve this task we use the round brackets <code>()</code>.
+If you want to find either <code>"Penguin"</code> or <code>"Pumpkin"</code> in a string, you could use the following Regular Expression: <code>/P(engu|umpk)in/g</code>
+Then you can test the code using the <code>test()</code> method to find whether the requires character(s)/string(s) are in the test string or not.
+<blockquote>let testStr = "Pumpkin";<br>let testRegex = /P(engu|umpk)in/g;<br>testRegex.test(testStr);<br>// Returns true</blockquote>
+</section>
+
+## Instructions
+<section id='instructions'>
+Fix the RegEx so that it checks for either <code>Hello</code> or <code>Halo</code> in a case sensitive manner.
+Then fix the code so that the RegEx that you have created is checked against <code>myString</code> and either <code>true</code> or <code>false</code> is returned depending on whether the RegEx matches.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: Your regex should return <code>true</code> for the string <code>"Hello"</code>.
+    testString: assert(myRegex.test('Hello'), 'Your regex should return <code>true</code> for the string <code>"Hello"</code>.');
+  - text: Your regex should return <code>true</code> for the string <code>"Halo"</code>.
+    testString: assert(myRegex.test('Halo'), 'Your regex should return <code>true</code> for the string <code>"Halo"</code>.');
+  - text: Your regex should return <code>false</code> for the string <code>"Hell"</code>.
+    testString: assert(!myRegex.test('Hell'), 'Your regex should return <code>false</code> for the string <code>"Hell"</code>.');
+  - text: You should use <code>test()</code> to test the regex.
+    testString: assert(code.match(/myRegex.test\(.*\), 'You should use <code>test()</code> to test the regex.');
+  - text: Your result should return <code>true</code>.
+    testString: assert(result === true, 'Your result should return <code>true</code>.');
+```
```

Comment:
These tests are fine in the sense of the description, but they leave the exercise contrived — there's no need for grouping to solve this challenge with the seed string — Hello|Halo does it. If we care enough, this is solvable by asking them to match against something much longer, so the bruteforce solution would produce a very long and ugly regex.

Reply:
What do you suggest we ask the user to check for?

Reply:
This is why I was checking the regex the user had written

Reply:
You mean the dot with the .test()? I don't know, without the dot, the test was failing maybe something else and refer here once: https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/using-the-test-method.english.md that has been written with the . too. Maybe you can make a PR for that?

Reply:
The test is failing because of something else. You can even check the results of this run to see for yourself.

Anyway, since `test` is already taught separately, why bother here? Just make this part of the seed code, so users focus on the regex.

Reply:
Removing the test for test then, I've made changes to the whole thing. Have a look